python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -110,33 +125,61 @@ def observedIndices(self):
         bandPassNames = self.obs_metadata.bandpass
         return [self.lsstBandpassDict.keys().index(x) for x in bandPassNames]
 
-    def get_snid(self):
-        # Not necessarily unique if the same galaxy hosts two SN
-        # Use refIdCol to access the relevant id column of the dbobj
-        # Should revert to galTileID for galaxyTiled catalogDBObj and
-        # id for galaxyObj catalogDBObj
-        # (email from Scott)
-        return self.column_by_name(self.refIdCol)
+    @compound('TsedFilepath', 'TmagNorm')
+    def get_phosimVars(self):
+        """
+        Obtain variables TsedFilepath to be used to obtain unique filenames
+        for each SED for phoSim and TMagNorm which is also used.
+        """
+        # construct the unique filename
+        # method: snid_mjd(to 4 places of decimal)_bandpassname
+        mjd = "_{:0.4f}_".format(self.mjdobs)
+        mjd += self.obs_metadata.bandpass + '.dat'
+        fnames = np.array([self.prefix + 'specFile_' + str(int(elem)) + mjd
+                  for elem in self.column_by_name('snid')], dtype='str')
+
+        c, x1, x0, t0, z  = self.column_by_name('c'),\
+                         self.column_by_name('x1'),\
+                         self.column_by_name('x0'),\
+                         self.column_by_name('t0'),\
+                         self.column_by_name('redshift')
+
+        bp = Bandpass()
+        bp.imsimBandpass()
+
+        magNorms = np.zeros(len(fnames))
+
+        snobject = SNObject()
+        snobject.rectifySED = True
+        for i in range(len(self.column_by_name('snid'))):
+            # if t0 is nan, this was set by the catalog for dim SN, or SN
+            #   outside redshift range, We will not provide a SED file for these
+            if np.isnan(t0[i]):
+                magNorms[i] = np.nan
+                fnames[i] = None
 
-    @property
-    def numobjs(self):
-        return len(self.column_by_name('id'))
+            else:
+                snobject.set(c=c[i], x1=x1[i], x0=x0[i], t0=t0[i],
+                             z=z[i])
+                if snobject.modelOutSideTemporalRange == 'zero':
+                    if self.mjdobs > snobject.maxtime() or self.mjdobs < snobject.mintime():
+                        magNorms[i] = np.nan
+                        fnames[i] = None
 
+                # SED in rest frame
+                sed = snobject.SNObjectSourceSED(time=self.mjdobs)
+                try:
+                    magNorms[i] = sed.calcMag(bandpass=bp)
+                except:
+                    # sed.flambda = 1.0e-20 
+                    magNorms[i] = 1000. # sed.calcMag(bandpass=bp)
```

Comment:
Should you also set `fnames[i]=None` in this case?  That way, the PhoSim InstanceCatalog will skip over this supernova, since `sedFilepath` is in the `cannot_be_null` member variable of the PhoSim InstanceCatalog.